packages/eslint-plugin/src/rules/unbound-method.ts

```diff
@@ -305,7 +305,14 @@ function isSafeUse(node: TSESTree.Node): boolean {
       return ['instanceof', '==', '!=', '===', '!=='].includes(parent.operator);
 
     case AST_NODE_TYPES.AssignmentExpression:
-      return parent.operator === '=' && node === parent.left;
+      return (
+        parent.operator === '=' &&
+        (node === parent.left ||
+          (node.type === AST_NODE_TYPES.MemberExpression &&
```

Comment:
technically this won't handle `this.foo.bar` and `super.foo.bar`, but we can ignore those cases until someone complains.